include/tracelogging/TraceLoggingProvider.h

```diff
@@ -2642,50 +2599,64 @@ ctype const*)).
     _tlg_data[_tlg_idx++] = lttngh_DataDescCreateSequenceUtf##NN((char##NN##_t const *)_tlg_temp##n, (cchValue));
 #define _tlg_DataDescCreate_tlg_Binary(n, ctype, pValue, cbValue, ndt)                                                \
     ctype const *const _tlg_temp##n = (pValue);                                                                       \
-    _tlg_DataDescCreateArray(&_tlg_data[_tlg_idx], _tlg_temp##n, (cbValue), sizeof(uint8_t), lttng_alignof(uint8_t)); \
+    _tlg_DataDescCreateArray(&_tlg_data[_tlg_idx], _tlg_temp##n, (cbValue), sizeof(uint8_t), lttngh_ALIGNOF(uint8_t)); \
     _tlg_idx += 2;
 #define _tlg_DataDescCreate_tlg_Sid(n, ctype, pValue, ndt)                                                                             \
     ctype const *const _tlg_temp##n = (pValue);                                                                                        \
-    _tlg_DataDescCreateArray(&_tlg_data[_tlg_idx], _tlg_temp##n, _tlg_SidSize(_tlg_temp##n), sizeof(uint8_t), lttng_alignof(uint8_t)); \
+    _tlg_DataDescCreateArray(&_tlg_data[_tlg_idx], _tlg_temp##n, _tlg_SidSize(_tlg_temp##n), sizeof(uint8_t), lttngh_ALIGNOF(uint8_t)); \
     _tlg_idx += 2;
-#define _tlg_DataDescCreate_tlg_GuidPtr(n, ctype, pValue, name)                                                                       \
+#define _tlg_DataDescCreate_tlg_ActivityId(n, ctype, pValue, name)                                                                       \
     ctype const *const _tlg_temp##n = (pValue);                                                                                       \
-    _tlg_DataDescCreateTinyArray(&_tlg_data[_tlg_idx], _tlg_temp##n, _tlg_temp##n ? 16 : 0, sizeof(uint8_t), lttng_alignof(uint8_t)); \
+    _tlg_DataDescCreateTinyArray(&_tlg_data[_tlg_idx], _tlg_temp##n, _tlg_temp##n ? 16 : 0, sizeof(uint8_t), lttngh_ALIGNOF(uint8_t)); \
     _tlg_idx += 2;
 #define _tlg_DataDescCreate_tlg_Buffer(n, ctype, pValue, ndt)                                                                            \
     ctype const *const _tlg_temp##n = (pValue);                                                                                          \
-    _tlg_DataDescCreateArray(&_tlg_data[_tlg_idx], _tlg_temp##n->Buffer, _tlg_temp##n->Length, sizeof(uint8_t), lttng_alignof(uint8_t)); \
+    _tlg_DataDescCreateArray(&_tlg_data[_tlg_idx], _tlg_temp##n->Buffer, _tlg_temp##n->Length, sizeof(uint8_t), lttngh_ALIGNOF(uint8_t)); \
     _tlg_idx += 2;
+
 #endif // _cplusplus
 
+#if lttngh_UST_VER >= 213
+#define _tlg_InitTracepoint(_tlg_fullName) lttngh_INIT_TRACEPOINT(NULL, _tlg_fullName)
+#define _tlg_EventFieldArray(eventFieldCount, ...) \
+    _tlg_FOREACH(_tlg_EventField, __VA_ARGS__) \
+    static lttngh_ust_event_field const* const _tlg_eventFieldArray[eventFieldCount] = { \
+        _tlg_FOREACH(_tlg_EventFieldRef, __VA_ARGS__) };
```

Comment:
In LTTNG 2.13, fields are stored as "array of pointer to struct", so our code for the fields is generated in two steps.
```C
// Step 1:
static const FieldInfo _tlgField1 = { ... };
static const StuffNeededForField2 _tlgField2_stuff = { ... };
static const FieldInfo _tlgField2 = { ..., &_tlgField2_stuff };
static const FieldInfo _tlgField3 = { ... };
// Step 2:
static const FieldInfo const* _tlgFields[3] = { &_tlgField1, &_tlgField2, &_tlgField3 };
```